frontend-mobile/lib/view/coffee_call_screen.dart

```diff
@@ -0,0 +1,74 @@
+import 'package:coffee_caller/communication/coffee_caller_protocol.dart';
+import 'package:coffee_caller/communication/socket_client.dart';
+import 'package:coffee_caller/view/cubit/coffee_call_cubit.dart';
+import 'package:coffee_caller/view/cubit/coffee_call_state.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+class CoffeeCallScreen extends StatelessWidget {
+  final SocketClient socketClient;
+
+  const CoffeeCallScreen({
+    super.key,
+    required this.socketClient,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('Coffee Caller'),
+      ),
+      body: BlocProvider(
+        create: (_) => CoffeeCallCubit(
+          socket: socketClient,
+          protocol: CoffeeCallerProtocol(socketClient: socketClient),
+        )
+          ..init()
+          ..connect(),
+        child: const CoffeeCallBody(),
+      ),
+    );
+  }
+}
+
+class CoffeeCallBody extends StatelessWidget {
+  const CoffeeCallBody({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        Row(
+          children: [
+            TextButton(
+              onPressed: context.read<CoffeeCallCubit>().join,
+              child: const Text('Join'),
+            ),
+            TextButton(
+              onPressed: context.read<CoffeeCallCubit>().leave,
+              child: const Text('Leave'),
+            ),
+            TextButton(
+              onPressed: context.read<CoffeeCallCubit>().start,
+              child: const Text('Start'),
+            ),
+          ],
+        ),
+        BlocBuilder<CoffeeCallCubit, CoffeeCallState>(
+          builder: (context, state) {
+            return ListView.builder(
```

Comment:
Is that the default list implementation of flutter? I would have expected something like a foreach

Reply:
The difference between a list view and a column is that the listview only builds the necessary items (hence the `builder` callback and not a `children` parameter) and has scrolling built in.